### ISSUE TITLE
error: unrecognised flag '--extra-experimental-features'

### ISSUE THREAD

**Issue by cawilliamson** [divnix/devos#406]
I am trying to re-setup devos again on my system after dumping flakes for a while and am having a tricky time achieving it.

So far I've faced and overcome a bunch of problems but have overcome them with help on both open issues and by merging this in to my code: https://github.com/divnix/devos/pull/403

However, the issue I'm facing right now I just can't seem to fix.

## Expected Behavior
A build should start allow me to switch to if by using switch instead if I wish to.

## Current Behavior
```
[nixos]$ bud rebuild thinkpad test
error: unrecognised flag '--extra-experimental-features'
Try 'nix --help' for more information.
```

## Steps to Reproduce
1. Git clone devos down
2. Merge in PR: https://github.com/divnix/devos/pull/403 (excluding flake.lock - generate that myself)
3. Attempt to build any host

## Context
I am completely unable to build any host configs until I find out how to resolve this issue. :(

**Comment by Pacman99** [comment]
So I think this might be because you have the wrong version of nix in your system. If your in nixos currently, did you set `nix.package` to `nixUnstable`/`nixFlakes`? 
This is my way of reproducing your issue:
```sh
$ nix shell nixpkgs#nix_2_3
$ nix --extra-experimental-features
error: unrecognised flag '--extra-experimental-features'
Try 'nix --help' for more information.
```

You can also try hopping into a shell with nix 2.4 before running bud, with `nix shell nixpkgs#nixUnstable`, but I'm not sure how bud finds what nix binary to use, so I don't know how effective a shell will be.

**Comment by water-sucks** [comment]
I think it's a problem with bud not finding the proper Nix version. I have exactly the same rev of nixUnstable and ran into the same issue even when doing it in a shell with `nix shell nixpkgs#nixUnstable`; however, I did find a workaround that allowed me to build it. I changed:
```nix
nixos.url = "github:nixos/nixpkgs/release-21.11";
latest.url = "github:nixos/nixpkgs/nixos-unstable";
```
to:
```nix
nixos.url = "github:nixos/nixpkgs/nixos-unstable";
latest.url = "github:nixos/nixpkgs/nixpkgs-unstable";
```
and ran `bud update`, and then the switch command worked. I did it out of a fresh clone of devos, though.
